README.md
<div>
  <span>
  <a href="http://www.azmag.gov/">
    <img
      alt="MAG"
      src="https://maps.azmag.gov/app/images/logos/MAG-logo.png"
      width="200"
    />
   </a>
  </span>
  <span>
  <a href="https://geo.azmag.gov/maps/azdemographics/">
    <img
      alt="Neighborhood Explorer"
      src="https://maps.azmag.gov/app/images/viewer-icons/icon_statewide-demographics.png"
      width="130"
    />
  </a>
  </span>
</div>

# Arizona Demographics

Map viewer for selected census info for Arizona

[![Website](https://img.shields.io/website-up-down-green-red/http/shields.io.svg?label=my-website)](http://geo.azmag.gov/maps/azdemographics/)
[![GitHub version](https://badge.fury.io/gh/AZMAG%2fmap-Demographic-Statewide.svg)](https://badge.fury.io/gh/AZMAG%2fmap-Demographic-Statewide)
[![GitHub issues](https://img.shields.io/github/issues/AZMAG/map-Demographic-Statewide.svg)](https://github.com/AZMAG/map-Demographic-Statewide/issues)
[![dependencies](https://david-dm.org/AZMAG/map-Demographic-Statewide.png)](https://david-dm.org/AZMAG/map-Demographic-Statewide)
[![devDependency Status](https://david-dm.org/AZMAG/map-Demographic-Statewide/dev-status.png)](https://david-dm.org/AZMAG/map-Demographic-Statewide)
[![Semver](http://img.shields.io/SemVer/2.0.0.png)](http://semver.org/spec/v2.0.0.html)
[![Built with Grunt](http://cdn.gruntjs.com/builtwith.png)](http://gruntjs.com/)
[![MIT licensed](https://img.shields.io/badge/license-MIT-blue.svg)](https://opensource.org/licenses/MIT)

Arizona Demographics is one of the Maricopa Association of Governments interactive mapping and analysis sites, showing selected population and housing data. The viewer allows you to graphically view selected population and housing data for all Census block groups for the State of Arizona. The viewer provides interactive mapping and reporting tools to allow exploration of data from Census 2010 and the American Community Survey (ACS) 2014-2018.

Use this site to explore Explore data from Census 2010 and American Community Survey (ACS) 2014-2018.

This web app viewer was developed by MAG staff for MAG member agencies and the public using ESRI's JavaScript API.

## Keywords

`Demographics` `Census` `2010` `American Community Survey` `ACS` `2014-2018` `5yr` `Counties` `State` `Arizona` `Maps` `MAG` `ESRI` `#MAGmaps`

## Version

Releases will be numbered with the following format:

**`<major>.<minor>.<patch>`**

And constructed with the following guidelines:

1. **MAJOR** version when you make incompatible API changes **bumps the major** resets minor and patch
2. **MINOR** version when you add functionality in a backwards-compatible manner **bumps the minor** resets patch
3. **PATCH** version when you make backwards-compatible bug fixes and misc changes **bumps only the patch**

### version | 4.0.8 ###

* Updated | 2019-12-03

* Created | 2014-10-21

## Credits

Maricopa Association of Governments (MAG) and the MAG member agencies

## Site URL

-   [Arizona Demographics](http://geo.azmag.gov/maps/azdemographics/)

## Documentation

A detailed list of the changes and fixes in each version update can be found in the CHANGELOG.md.

-   [CHANGELOG](CHANGELOG.md)

## Resources

## Examples

## Dependencies

-   [git-labelmaker](https://github.com/himynameisdave/git-labelmaker?utm_source=hashnode.com)

## Legal Disclaimer

The Maricopa Association of Governments provides the data within these pages as a public resource of general information for use "as is." The Maricopa Association of Governments GIS (Geographic Information System) departments provides this information with the understanding that it is not guaranteed to be accurate, correct or complete and any conclusions drawn from such information are the sole responsibility of the user. Further, the Maricopa Association of Governments GIS departments makes no warranty, representation or guaranty as to the content, sequence, accuracy, timeliness or completeness of any of the spatial or database information provided herein. While every effort has been made to ensure the content, sequence, accuracy, timeliness or completeness of materials presented within these pages, the Maricopa Association of Governments GIS Departments assumes no responsibility for errors or omissions, and explicitly disclaims any representations and warranties, including, without limitation, the implied warranties of merchantability and fitness for a particular purpose. The Maricopa Association of Governments GIS Departments shall assume no liability for:

    1. Any errors, omissions, or inaccuracies in the information provided, regardless of how caused; or
    2. Any decision made or action taken or not taken by viewer in reliance upon any information or data furnished hereunder.

Availability of the Maricopa Association of Governments Map Server is not guaranteed. Applications, servers, and network connections may be unavailable at any time for maintenance or unscheduled outages. Outages may be of long duration. Users are cautioned to create dependencies on these services for critical needs.

THE FOREGOING WARRANTY IS EXCLUSIVE AND IN LIEU OF ALL OTHER WARRANTIES OF MERCHANTABILITY, FITNESS FOR PARTICULAR PURPOSE AND/OR ANY OTHER TYPE WHETHER EXPRESSED OR IMPLIED. In no event shall Maricopa Association of Governments become liable to users of these data, or any other party, for any loss or direct, indirect, special, incidental or consequential damages, including, but not limited to, time, money or goodwill, arising from the use or modification of the data.

To assist the Maricopa Association of Governments in the maintenance and/or correction of the data, users should provide the Maricopa Association of Governments GIS Departments with information concerning errors or discrepancies found in using the data. Please use the e-mail contact address at the bottom of the affected web page.

Please acknowledge the Maricopa Association of Governments (MAG) GIS as the source when Map Server data is used in the preparation of reports, papers, publications, maps, or other products.

## Copyright and Licensing

Code released under the MIT license.

A copy of the license is available in the repository's `LICENSE` file.

-   [LICENSE](LICENSE)

## Tests

Tests use [Intern](https://theintern.io/) 

More information on running and writing tests can be found in [TESTS.md.](test/TESTS.md)

## Build

You can build the compiled files by running the default grunt task (`grunt`) from the root folder. This will compile (bundle and minify) the files and place them in the **build** folder. Running the `grunt run` task will run the tests and spin up a node server to test the application, watch for modified source code, and refresh after changes are saved.

Additionally, there is a `grunt deploy` task that will build the application *and* spin up a node server to test the built files.

CHANGELOG.md
# CHANGELOG #
#### State Demographics Map Viewer ####

[![Semver](http://img.shields.io/SemVer/2.0.0.png)](http://semver.org/spec/v2.0.0.html)

All notable changes to this project will be documented in this file.

Releases will be numbered with the following format:

**`<major>.<minor>.<patch>`**

And constructed with the following guidelines:

1. **MAJOR** version when you make incompatible API changes **bumps the major** resets minor and patch
2. **MINOR** version when you add functionality in a backwards-compatible manner **bumps the minor** resets patch
3. **PATCH** version when you make backwards-compatible bug fixes and misc changes **bumps only the patch**

**************************************************

#### Types of Changes ####

* `Added` for new features
* `Changed` for changes in existing functionality
* `Deprecated` for soon-to-be removed features
* `Removed` for now removed features
* `Fixed` for any bug fixes
* `Security` in case of vulnerabilities

[Source](http://keepachangelog.com/en/1.0.0/)

***************************************************

## Version 4.0.7 | 2019-07-24

* Fix the order of Report drop down to move COGs & MPOs to bottom [github #260]
* Fix the order of the layers list to move COGs & MPOs to bottom

## Version 4.0.7 | 2019-07-24

* Changed language in custom classification types in the maps panel
* Fixed build system

## Version 4.0.6 | 2019-04-02

* Changed Layer from "Council" to Councils of Governments

## Version 4.0.5 | 2019-03-28

* Fixed widget hover titles - style issue

## Version 4.0.4 | 2019-03-28

* Fixed Print widget style issues
* Added JPG icon to print widget when selected

## Version 4.0.3 | 2019-03-28


## Version 4.0.2 | 2019-03-19

* Fixed Share Funcationality
* Reordered TOC
* Added Spatial Layers to Advanced Selection Report
* Added footnote for Minority Population
* Updated SVMPO Contact info

## Version 4.0.1 | 2019-02-20

* Added Print widget

## Version 4.0.0 | 2019-02-13

* Changed Build
* Added Normalize.css
* Complete rewrite of code in 4.x
* Migrated code from bikeways viewer and from existing demographics viewer as possible

## Version 3.2.4 | 2018-04-20

* updated css for header and menu bars
* update logo
* Removed yellow selection graphic
* Update logo link to new website
* Update SVMPO Contact info
* Add Non-Hispanic to the acs report.
* Fixed the query builder with percents.
* Update Verbiage in Query Builder Help window.
* Update verbiage in Census tabs for non standard geometries.
* Fixed the legend loading issue in which the block groups legend would sometimes not show the class breaks correctly.  It works every time now.
* Updated custom summary to correct data source when user activates the "Census Block Groups" tab.
* Move Run Query and Cancel buttons to bottom right of Query Window.  Remove unnecessary queryBuilder Two view.
* Update column header in Title VI Data Tab
* Updated legal Census Disclaimer
* Updated cbr config to include normalization fields.  This allows the percent to be calculated on the fly.
* Updated Demographic Config PDF Links to https to prevent issue with pdf exports.
* Added some Google analytics events.  Updated the grunt-file to remove comments.

## Version 3.2.3 | 2018-02-26

* fixed bug in data query found in training class
* updated grunt files

## Version 3.2.2 | 2018-02-15

* Update Title VI data to only appear in areas that include  > 5000 Total Population.
* Made print label longer
* Oops, my legendvm refactoring broke the multiple map frames feature.  This is the fix.
* Add Error log if config item doesn't match up correctly
* Refactored Legend code to make much simpler going forward.
* Update Help view
* Updated legal disclaimer for census tabs.
* Updated the buffer selection to prevent the error if a user enters 0 in the distance box.  Also added a max of 10,000 to the distance box.
* Update Data Source for Advanced Query Help
* Fixed issues with summary report help items
* Fixed typo.  Debbie found this.
* Changing resources to https
* Fixed bug in which datasource becomes undefined in multiple map frames
* Refactored ExportToPDF function and updated PDF links.
* Updated some verbage and fields for the title 6 data.

## Version 3.2.1 | 2018-01-18

* Fixed the help contact popup window.
* Fixed the field duplication issue for all summary reports.
* Fixed the search functionality.  Added new report types to the search.
* Updated the popups for supervisor and city council districts.
* Reduced the opacity to make it easier to see the colors beneath the selection graphic.
* Updated help windows to include supervisor and council district verbage.
* Added Use Statewide Legend checkbox.  Fixed a bug that was making the legend refresh twice.
* Updated the Advanced Query Window to point to relative urls from demographic config.
* Added an alert that prevents a user from submitting a 0 results advanced query.
* Fixed bug in which the ACS data tab was sometimes being removed.

## Version 3.2.0 | 2018-01-18

* Added Supervisor Districts to Reports and layers
* Added City Council Districts to Reports and layers
* Refactored config layers to use the "layerName" property to match the rest endpoint layer instead of having the index hardcoded.
* Switched the Generate Renderer task to the client using the geostats library instead of the Generate Renderer GP tool.  This will increase speed and allow for dynamic renderering.

## Version 3.1.0 | 2017-05-23

* file changes only!
* NO code changes!
* removed node_modules from git
* added src folder
* added build folder

## Version 3.0.6 | 04/13/2017

* Fixed bugs related to the markup tools. (multiple edit tools were being instantiated, popup was showing when drawing.)
* Added fields back into the CBR View Model.  This allows them to be seen in the map and in the advanced query.
* Added area fields back into the advanced query.

## Version 3.0.5 | 03/31/2017

* fix paging issue in custom summary

## Version 3.0.4 | 03/23/2017


src/config.js
/**
 *  layerInfo:
 *       id = layer unique identifier
 *       title = display title as used in the interface
 *       category = layer category - displayed at top of each layer group
 *       url = layer url
 *       type = [dynamic | tile | wms] - type of layer definition
 *       historical = [true | false] - True if the layer is a historical layer
 *       version = (WMS only) WMS version number
 *       visible = [true | false] default layer visibility
 *       isBasemap = [true | false] layer is considered a basemap (displayed in basemap control)
 *       query = url used for query operations
 *       queryWhere = default where clause used for query operations
 *       layers = array of layer indicies (from MXD) activated when layer is checked "on"
 *       selectable = [true | false] layer is available in sketch dropdown list
 *       layerDefFields = array of fields used for definition query text inputs
 *           - type = ['list'] type of input
 *           - label = label displayed to the user
 *           - fields = array of fields used in query where clause (ex: {name: 'FIELDNAME', numeric:[true|false]} )
 *           - list = array of dropdown list items (ex: {"id": 0, "OCEAN": 'AL', "YEAR": 2012, "DEPRESSION": 18, "LABEL": 'Sandy'})
 *               - Must have property entry for each object in fields array.  'label' property is reserved for display to user
 *           - numeric = [true | false] indicates whether field is numeric or not
 *       opacity = default layer opacity
 *       token = token used for secure service access
 *       filters = array of layer filters / sublayers
 *           - category = filter category (title displayed for each group)
 *           - type = [exp | layer]
 *           - name = label displayed next to filter checkbox
 *           - layers = array of layer indicies (MXD) associated with checkbox
 *           - expression = definition expression associated with filter
 *           - visible = default selection / visibility of filter
 **/

var appConfig = new function() {

    this.Version = "v3.2.5 | 2018-07-31";
    this.copyright = "2018";
    this.LegendSource = "American Community Survey 2012-2016 5yr";
    this.jasonemail = "https://www.azmag.gov/Contact/4788?s=geo.azmag.gov/maps/azdemographics&n=Jason%20Howard&popUp=true";

    this.ArcGISInstanceURL = "https://geo.azmag.gov/gismag/rest";
    //this.exportWebMapUrl = "http://geo.azmag.gov/gismag/rest/services/Utilities/PrintingTools/GPServer/Export%20Web%20Map%20Task";  // Generic Print Service
    this.exportWebMapUrl = "https://geo.azmag.gov/gismag/rest/services/gp/stateDemo/GPServer/Export%20Web%20Map"; // Custom Print Service
    this.webServicePasscode = "sun sand dry heat grand canyon";

    this.mainURL = "https://geo.azmag.gov/gismag/rest/services/maps/DemographicState2016/MapServer";
    this.siteUrl = "https://geo.azmag.gov/maps/azdemographics/?";

    // Search Service URLs
    this.geoCoderService = "//geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer";
    this.tractService = this.mainURL + "/23";
    this.SchoolsService = this.mainURL + "/25";

    this.layerInfo = [{
        drawOrder: 0,
        id: "censusTracts",
        title: "Census Tracts",
        type: "dynamic",
        url: this.mainURL,
        queryUrl: this.mainURL + "/24",
        queryWhere: "1=1",
        layers: [23, 24],
        opacity: 1,
        visible: false,
        showTOC: true,
        showLegend: true,
        link: false
    }, {
        drawOrder: 2,
        id: "cogBoundaries",
        layerName: "Cogs_Mpos",
        title: "COG / MPO Boundaries",
        type: "feature",
        queryWhere: "1=1",
        opacity: 0.7,
        visible: false,
        selectable: true,
        outFields: ["*"],
        filters: [],
        showTOC: true,
        link: true
    }, {
        drawOrder: 1,
        id: "countyBoundaries",
        title: "County Boundaries",
        layerName: "Counties",
        type: "feature",
        queryWhere: "1=1",
        opacity: 0.8,
        visible: true,
        selectable: true,
        outFields: ["*"],
        filters: [],
        showTOC: true,
        link: true
    }, {
        drawOrder: 3,
        layerName: "Congressional_Districts",
        id: "congressionalDistricts",
        title: "Congressional Districts",
        type: "feature",
        queryWhere: "1=1",
        opacity: 1,
        visible: false,
        selectable: true,
        outFields: ["*"],
        filters: [],
        showTOC: true,
        link: true
    }, {
        drawOrder: 4,
        layerName: "Legislative_Districts",
        id: "legislativeDistricts",
        title: "Legislative Districts",
        type: "feature",
        queryWhere: "1=1",
        opacity: 1,
        visible: false,
        selectable: true,
        outFields: ["*"],
        filters: [],
        showTOC: true,
        link: true
    }, {
        drawOrder: 5,
        layerName: "Zip_Codes",
        id: "zipCodes",
        title: "ZIP Codes",
        type: "feature",
        queryWhere: "1=1",
        opacity: 1,
        visible: false,
        selectable: true,
        outFields: ["*"],
        filters: [],
        showTOC: true,
        link: true
    }, {
        drawOrder: 6,
        id: "districts",
        title: "Unified School Districts",
        url: this.mainURL + "/25",
        queryUrl: this.mainURL + "/25",
        type: "feature",
        queryWhere: "1=1",
        opacity: 1,
        visible: false,
        selectable: true,
        outFields: ["*"],
        filters: [],
        showTOC: true,
        link: false
    }, {
        drawOrder: 7,
        layerName: "SupervisorDistricts",
        id: "supervisorDistricts",
        title: "Supervisor Districts",
        type: "feature",
        queryWhere: "1=1",
        opacity: 1,
        visible: false,
        selectable: true,
        outFields: ["*"],
        filters: [],
        showTOC: true,
        link: false
    }, {
        drawOrder: 7,
        layerName: "CityDistricts",
        id: "councilDistricts",
        title: "Council Districts",
        type: "feature",
        queryWhere: "1=1",
        opacity: 1,
        visible: false,
        selectable: true,
        outFields: ["*"],
        filters: [],
        showTOC: true,
        link: false
    }, {
        drawOrder: 7,
        id: "esriReference",
        title: "Streets",
        type: "tile",
        url: "http://server.arcgisonline.com/arcgis/rest/services/Reference/World_Transportation/MapServer",
        visible: true,
        showTOC: true,
        showLegend: false
    }, {
        drawOrder: 8,
        id: "blockGroups",
        title: "American Community Survey by Block Group, 2012-2016 5yr",
        type: "dynamic",
        url: this.mainURL,
        layers: [0],
        opacity: 0.8,
        visible: true,
        showTOC: false,
        link: false,
        populateDropDown: false
    }, {
        drawOrder: 9,
        id: "esriBasemap",
        title: "Terrain",
        type: "tile",
        // url: "http://server.arcgisonline.com/ArcGIS/rest/services/World_Terrain_Base/MapServer",
        // url: "http://server.arcgisonline.com/arcgis/rest/services/World_Street_Map/MapServer",
        // url: "http://server.arcgisonline.com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer",
        url: "http://server.arcgisonline.com/arcgis/rest/services/Canvas/World_Light_Gray_Base/MapServer",
        visible: true,
        isBasemap: false,
        showTOC: false,
        showLegend: false
    }, {
        drawOrder: 10,
        id: "esriImagery",
        title: "Imagery",
        type: "tile",
        url: "http://services.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer",
        isBasemap: true,
        visible: false,
        showTOC: true,
        showLegend: false
    }];

    // function toObject(arr) {
    //     var rv = {};
    //     for (var i = 0; i < arr.length; ++i)
    //         rv[arr[i].id] = arr[i];
    //     return rv;
    // }


    // this.layerInfoObj = toObject(this.layerInfo);
    // End layerInfo

    this.initExtent = {
        "xmin": -13271172.93,
        "ymin": 3506737.09,
        "xmax": -11501054.53,
        "ymax": 4612403.73,
        "spatialReference": {
            "wkid": 102100
        }
    };

    //from colorbrewer 2.0 qualitative HEX Paired
    this.seriesColors = ["#A6CEE3", "#1F78B4", "#B2DF8A", "#33A02C", "#FB9A99", "#E31A1C", "#FDBF6F", "#FF7F00", "#CAB2D6", "#6A3D9A", "#FFFF99", "#B15928"];
    //from colorbrewer 2.0 qualitative HEX Set3
    //this.seriesColors = ["#8DD3C7", "#FFFFB3", "#BEBADA", "#FB8072", "#80B1D3", "#FDB462", "#B3DE69", "#FCCDE5", "#D9D9D9", "#BC80BD", "#CCEBC5", "#FFED6F"];

    this.bubbleColors = [{
        category: "MgBizFin",
        color: "#A6CEE3"
    }, {
        category: "CompEngSci",
        color: "#1F78B4"
    }, {
        category: "EduLegComArtMedia",
        color: "#B2DF8A"
    }, {
        category: "HealthTechnical",
        color: "#33A02C"
    }, {
        category: "ServiceOcc",
        color: "#FB9A99"
    }, {
        category: "ProtectiveServ",
        color: "#E31A1C"
    }, {
        category: "SalesOfficeOcc",
        color: "#FDBF6F"
    }, {
        category: "NatResources",
        color: "#FF7F00"
    }, {
        category: "ProdTransMaterial",
        color: "#CAB2D6"
    }];

    this.URLMinimizer = {

        login: "vwolfley",
        apiKey: "R_8dbab4a2f0664e8f8b4f88fe0d9d7f80"
    };

    // ------------------------------
    // MARKUP TOOL SETTINGS
    // ------------------------------

    // Specify the Markup / drawing tools.
    this.markupToolTreeNodes = [{
        id: 1,
        text: "Polygon",
        DisplayText: "Polygon",
        Type: "POLYGON",
        imageUrl: "app/resources/img/i_draw_poly.png"
    }, {
        id: 2,
        text: "Circle",
        DisplayText: "Circle",
        Type: "CIRCLE",
        imageUrl: "app/resources/img/i_draw_circle.png"
    }, {
        id: 3,
        text: "Arrow",
        DisplayText: "Arrow",
        Type: "ARROW",
        imageUrl: "app/resources/img/i_draw_arrow.png"
    }, {
        id: 4,
        text: "Freehand",
        DisplayText: "Freehand",
        Type: "FREEHAND_POLYGON",
        imageUrl: "app/resources/img/i_draw_freepoly.png"
    }, {
        id: 5,
        text: "Text",
        DisplayText: "Text Box",
        Type: "POINT",
        imageUrl: "app/resources/img/i_draw_text.png"
    }];

    // Specify the Markup Tool's Fill and Outline Kendo Color Pallettes.
    this.fillColorPalette = ["rgba(163, 73, 164, .50)", "rgba(63, 72, 204, .5)", "rgba(0, 162, 232, 0.50)",
        "rgba(34, 177, 76, 0.50)", "rgba(255, 242, 0, 0.50)", "rgba(255, 127, 39, 0.50)",
        "rgba(237, 28, 36, 0.50)", "rgba(136, 0, 21, 0.50)", "rgba(127, 127, 127, 0.50)", "rgba(0, 0, 0, 0.50)"
    ];
    this.fillColorOpacity = 0.75;

    this.outlineColorPalette = ["rgba(163, 73, 164, 1.0)", "rgba(63, 72, 204, 1.0)", "rgba(0, 162, 232, 1.0)",
        "rgba(34, 177, 76, 1.0)", "rgba(255, 242, 0, 1.0)", "rgba(255 ,127, 39, 1.0)",
        "rgba(237, 28, 36, 1.0)", "rgba(136, 0, 21, 1.0)", "rgba(127, 127, 127, 1.0)", "rgba(0, 0, 0, 1.0)"
    ];
    this.outlineColorOpacity = 1.0;

    this.textSymbolFontSizes = [8, 9, 10, 11, 12, 14, 16, 18, 20, 24, 30, 50];

    this.sourceLabel = "Source: United States Census Bureau, American Community Survey 2012-2016 5yr Estimates";

    this.sourceLabel2 = "Source: United States Census Bureau, 2010 Decennial Census";

    this.legalACSDisclaimer = 'Source: U.S. Census Bureau, 2012-2016 American Community Survey (ACS) 5-Year Estimates. ACS data are based on a sample and are subject to sampling variability.  The degree of uncertainty for an estimate is represented through the use of a margin of error (MOE).  In addition to sampling variability, the ACS estimates are subject to nonsampling error. The MOE and effect of nonsampling error is not represented in these tables. Supporting documentation on subject definitions, data accuracy, and statistical testing can be found on the American Community Survey website (www.census.gov/acs) in the Data and Documentation section. Sample size and data quality measures (including coverage rates, allocation rates, and response rates) can be found on the American Community Survey website (www.census.gov/acs) in the Methodology section.  The MOE for individual data elements can be found on the American FactFinder website (factfinder2.census.gov).  Note: Although the ACS produces population, demographic and housing unit estimates, the 2010 Census provides the official counts of the population and housing units for the nation, states, counties, cities and towns.  Prepared by: Maricopa Association of Governments, www.azmag.gov, (602) 254-6300';

    this.legalCensusDisclaimer = 'Source: United States Census Bureau, 2010 Decennial Census. Additional information about the 2010 Decennial Cenuss can be found on the US Census Bureau website at www.census.gov/2010census. Prepared by: Maricopa Association of Governments, www.azmag.gov, (602) 254-6300';
}; //End Config
